tools/baseimage/cmd/amend_gce_base_image/main.go

```diff
@@ -0,0 +1,172 @@
+// Copyright (C) 2025 The Android Open Source Project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"path/filepath"
+	"time"
+
+	"github.com/google/android-cuttlefish/tools/baseimage/pkg/gce"
+	"github.com/google/android-cuttlefish/tools/baseimage/pkg/gce/scripts"
+)
+
+const (
+	outImageName = "amended-image"
+)
+
+var (
+	project                    = flag.String("project", "", "GCP project whose resources will be used for creating the amended image")
+	zone                       = flag.String("zone", "us-central1-a", "GCP zone used for creating relevant resources")
+	source_image_project       = flag.String("source-image-project", "", "Source image GCP project")
+	source_image               = flag.String("source-image", "", "Source image name")
+	cuttlefish_debs_zip_source = flag.String("cuttlefish-debs-zip-source", "", "Local path to zip file containing cuttlefish debian packages")
+)
+
+type amendImageOpts struct {
+	SourceImageProject      string
+	SourceImage             string
+	CuttlefishDebsZipSource string
+}
+
+func installCuttlefishDebs(project, zone, insName, zipSrc string) error {
+	dstSrc := "/tmp/" + filepath.Base(zipSrc)
+	if err := gce.UploadFile(project, zone, insName, zipSrc, dstSrc); err != nil {
+		return fmt.Errorf("error uploading %s: %v", zipSrc, err)
+	}
+	if err := gce.UploadBashScript(project, zone, insName, "install.sh", scripts.InstallCuttlefishPackages); err != nil {
+		return fmt.Errorf("error uploading script: %v", err)
+	}
+	if err := gce.RunCmd(project, zone, insName, "./install.sh "+dstSrc); err != nil {
+		return err
+	}
+	return nil
+}
+
+func ammendImageMain(project, zone string, opts amendImageOpts) error {
+	h, err := gce.NewGceHelper(project, zone)
+	if err != nil {
+		return fmt.Errorf("failed to create GCE helper: %w", err)
+	}
+	insName := outImageName
+	attachedDiskName := fmt.Sprintf("%s-attached-disk", insName)
+	defer func() {
```

Comment:
Should this be done after checking `CreateDisk` succeeded? As it is now it will run even if `CreateDisk` fails.

Reply:
Done.